x-pack/test/apm_api_integration/tests/transactions/transactions_groups_comparison_statistics.ts

```diff
@@ -151,7 +152,72 @@ export default function ApiTest({ getService }: FtrProviderContext) {
 
         expect(response.status).to.be(200);
-        expect(response.body).to.empty();
+        expect(response.body).to.be.eql({ currentPeriod: {}, previousPeriod: {} });
+      });
+
+      it('returns data with previous period', async () => {
+        const response = await supertest.get(
+          url.format({
+            pathname: `/api/apm/services/opbeans-java/transactions/groups/comparison_statistics`,
+            query: {
+              uiFilters: '{}',
+              numBuckets: 20,
+              transactionType: 'request',
+              latencyAggregationType: 'avg',
+              transactionNames: JSON.stringify(transactionNames),
+              start: moment(end).subtract(15, 'minutes').toISOString(),
+              end,
+              comparisonStart: start,
+              comparisonEnd: moment(start).add(15, 'minutes').toISOString(),
+            },
+          })
+        );
+
+        expect(response.status).to.be(200);
+
+        const {
+          currentPeriod,
+          previousPeriod,
+        } = response.body as TransactionsGroupsComparisonStatistics;
+
+        expect(Object.keys(currentPeriod).length).to.be.eql(transactionNames.length);
+        expect(Object.keys(previousPeriod).length).to.be.eql(transactionNames.length);
+
+        transactionNames.map((transactionName) => {
+          expect(currentPeriod[transactionName]).not.to.be.empty();
+          expect(previousPeriod[transactionName]).not.to.be.empty();
+        });
+
+        const currentPeriodStatistics = currentPeriod[transactionNames[0]];
+        const previousPeriodStatistics = previousPeriod[transactionNames[0]];
+
+        expect(removeEmptyCoordinates(currentPeriodStatistics.latency).length).to.be.greaterThan(0);
+        expect(removeEmptyCoordinates(previousPeriodStatistics.latency).length).to.be.greaterThan(
+          0
+        );
+        expectSnapshot(currentPeriodStatistics.latency).toMatch();
+        expectSnapshot(previousPeriodStatistics.latency).toMatch();
+
+        expect(removeEmptyCoordinates(currentPeriodStatistics.throughput).length).to.be.greaterThan(
+          0
+        );
+        expect(
+          removeEmptyCoordinates(previousPeriodStatistics.throughput).length
+        ).to.be.greaterThan(0);
+        expectSnapshot(currentPeriodStatistics.throughput).toMatch();
+        expectSnapshot(previousPeriodStatistics.throughput).toMatch();
+
+        expect(removeEmptyCoordinates(currentPeriodStatistics.errorRate).length).to.be.greaterThan(
+          0
+        );
+        expect(removeEmptyCoordinates(previousPeriodStatistics.errorRate).length).to.be.greaterThan(
+          0
+        );
+        expectSnapshot(currentPeriodStatistics.errorRate).toMatch();
+        expectSnapshot(previousPeriodStatistics.errorRate).toMatch();
+
+        expectSnapshot(roundNumber(currentPeriodStatistics.impact)).toMatchInline(`"21.75"`);
+        expectSnapshot(roundNumber(previousPeriodStatistics.impact)).toMatchInline(`"96.94"`);
```

Comment:
This might be too many tests - do you think it would help if they were broken down in multiple tests?